File: tidyM365.R
# tidyM365.R is a function that cleans data from the Stops-MillStyleYarnBOMs.csv file.

tidyM365 <- function(){ 
        # Load necessary libraries
        library(dplyr); library(lubridate)

        # Read in data and set to local variable        
        m365_tbl <- tbl_df(read.csv("~/Desktop/Pendleton/CSVs/M365.csv", 
                           na.strings = "N/A"))
        
        # Cleans column names of dataset
        names(m365_tbl) <- tolower(names(m365_tbl))
        names(m365_tbl) <- sub("\\.", "", names(m365_tbl))
        
        # Format date 
        m365_tbl$duedate <- mdy(m365_tbl$duedate)
        m365_tbl$date <- mdy(m365_tbl$date)
        m365_tbl$millstyle <- as.character(m365_tbl$millstyle)
        
        # Removes records of experimental millstyles
        m365_tbl <- m365_tbl[!grepl("EX", m365_tbl$millstyle),]
        
        # Creates range feature by pulling range sequence contained in millstyle
        m365_tbl$range <- gsub("(^[^1-9]*)(\\d+)((-|[A-Z]).+$)", "\\2", m365_tbl$millstyle)

        # Selects and orders fields from the table
        m365_tbl %>% 
                select(range, millstyle, order, duedate, date, yds, loom) %>%
                print()
        
        # Returns table
        return(m365_tbl)
}

File: Python Scripts/tidyM365.py
# tidyM365.py is a function that cleans data from the Stops-MillStyleYarnBOMs.csv file.

def tidyM365():
	# Load Libraries
	import pandas as pd

	# Read in data and set to local variable
	M365_filename = "~/Desktop/Pendleton/CSVs/M365.csv"
	M365_df = pd.read_csv(M365_filename)

	# Cleans column names
	M365_df.columns = map(str.lower, M365_df.columns)
	M365_df.columns = M365_df.columns.str.replace(" ", "")

	# Removes experimental millstyles
	M365_df = M365_df[M365_df['millstyle'].str.contains("EX") == False]

	# Add range field
	M365_df['range'] = pd.Series()
	M365_df['range'] = M365_df['millstyle'].str.extract('^(.*?)(-|[A-Z]).*', expand = True)

	# Sets date field as datetime data type
	M365_df['date'] = pd.to_datetime(M365_df['date'])

	# Selects and orders dataframe fields
	M365_df = M365_df[['range', 'millstyle', 'order', 'duedate', 'date', 'yds', 'loom']]

	return M365_df